Openshift testing scripts and documentation

wallrj opened the pull request:
* A makefile which launches a Google Cloud VM, installes `crc` and then runs `crc setup` and `crc start`
* Documentation explaining the steps to install `crc` manually on a VM

maelvls commented on README.md:
````diff
@@ -128,3 +128,160 @@ Run some of the cert-manager E2E conformance tests:
+Download your pull secret from the [crc-download] page and supply the path in the command line below:
+
+```sh
+make -f hack/crc.mk crc-instance OPENSHIFT_VERSION=4.9 PULL_SECRET=~/Downloads/pull-secret
````
```suggestion
make -f hack/crc.mk crc-instance OPENSHIFT_VERSION=4.9 PULL_SECRET=$HOME/Downloads/pull-secret
```

wallrj commented:
Thanks for the suggestion @maelvls and for testing the script.

I've made few further changes:
 * Moved `make crc-instance` to the top-level Makefile
 * Added a `make crc-e2e` target which uploads a cert-manager E2E test binary and runs a subset of the E2E tests.

At this point, I'm going to merge this and make further improvements in followup PRs.
